src/android/SecureStorage.java

```diff
@@ -158,7 +158,7 @@ public void run() {
                     public void run() {
                         try {
                             byte[] decryptedKey = rsa.decrypt(encKey, service2alias(service));
-                            String decrypted = new String(AES.decrypt(ct, decryptedKey, iv, adata));
+                            String decrypted = new String(AES.decrypt(ct, decryptedKey, iv, adata, data.getString("mode")));
```

Comment:
I report your pull request into a common PR.
But the maintainer ask me more details : https://github.com/Sotam/cordova-plugin-secure-storage-android10/pull/1

Can you help me to document possible values for "mode" ?

Reply:
mode contains the Cipher Mode. It can be the default value of the plugin, but it's saved the data is encrypted, and stored as a string. Like "CBC" or "ECB" for example. When changing the default cipher, old data can still be decrypted using the right cipher with this pull request.

Reply:
Thanks for your anwser. I copy it here: https://github.com/Sotam/cordova-plugin-secure-storage-android10/pull/1